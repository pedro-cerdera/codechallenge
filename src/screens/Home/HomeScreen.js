import React, {useRef} from 'react';
import {Image} from 'react-native';
import {useSelector} from 'react-redux';
import {useNavigation} from '@react-navigation/native';

import Logo from '../../assets/images/logo.png';
import {
  Account,
  DefaultSpacingContainer,
  OperationsList,
  SafeAreaContainer,
} from '../../components';
import styles from './styles';

const HomeScreen = () => {
  const {ammount} = useSelector((store) => ({
    ammount: store.TransactionReducer.ammount,
  }));

  const navigation = useNavigation();

  const operations = useRef([
    {
      id: '1',
      icon: 'fa-plus-circle',
      title: 'Nova Transação',
      onPress: () => navigation.navigate('NewTransaction'),
    },
    {
      id: '2',
      icon: 'fa-list',
      title: 'Listar Transações',
      onPress: () => navigation.navigate('Transactions'),
    },
  ]).current;

  return (
    <SafeAreaContainer testID={'home-container'}>
      <DefaultSpacingContainer style={styles.container}>
        <Image source={Logo} />
        <Account
          testID={'account'}
          ammount={ammount}
          onBottomPress={() => navigation.navigate('Transactions')}
        />
        <OperationsList operations={operations} />
      </DefaultSpacingContainer>
    </SafeAreaContainer>
  );
};

export default HomeScreen;
