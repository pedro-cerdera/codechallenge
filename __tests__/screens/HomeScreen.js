import React from 'react';
import {render} from 'react-native-testing-library';
import {useSelector} from 'react-redux';
import HomeScreen from '../../src/screens/Home/HomeScreen';

jest.mock('../../src/components/Account/Account', () => 'Account');

jest.mock('../../src/assets/images/logo.png', () => ({
  uri: 'logo.png',
}));
jest.mock(
  '../../src/components/OperationsList/OperationsList',
  () => 'OperationsList',
);
jest.mock('react-redux', () => ({
  useSelector: jest.fn(),
}));

const mockNavigationPress = jest.fn();

jest.mock('@react-navigation/native', () => ({
  useNavigation: (name) => mockNavigationPress(name),
}));

const mockAppState = {
  TransactionReducer: {
    ammount: 0,
    transactions: [],
    isLoading: false,
    error: null,
  },
};

describe('Testing HomeScreen', () => {
  beforeEach(() => {
    useSelector.mockImplementation((callback) => {
      return callback(mockAppState);
    });
  });
  afterEach(() => {
    useSelector.mockClear();
  });

  test('Render all home screen components correctly', async () => {
    const {getByTestId} = render(<HomeScreen />);
    expect(getByTestId('account').props.ammount).toBe(0);
  });

  test('renders correctly', () => {
    const tree = render(<HomeScreen />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
