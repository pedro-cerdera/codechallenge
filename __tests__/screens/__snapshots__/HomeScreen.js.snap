// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing HomeScreen renders correctly 1`] = `
<RCTSafeAreaView
  emulateUnlessSupported={true}
  style={
    Array [
      Object {
        "backgroundColor": "#1C2124",
        "flexBasis": 0,
        "flexGrow": 1,
        "flexShrink": 1,
      },
    ]
  }
  testID="home-container"
>
  <View
    style={
      Array [
        Object {
          "flexBasis": 0,
          "flexGrow": 1,
          "flexShrink": 1,
          "marginBottom": 20,
          "marginLeft": 20,
          "marginRight": 20,
          "marginTop": 20,
        },
        Object {
          "alignItems": "center",
          "justifyContent": "space-between",
        },
      ]
    }
  >
    <Image
      source={
        Object {
          "uri": "logo.png",
        }
      }
    />
    <Account
      ammount={0}
      onBottomPress={[Function]}
      testID="account"
    />
    <OperationsList
      operations={
        Array [
          Object {
            "icon": "fa-plus-circle",
            "id": "1",
            "onPress": [Function],
            "title": "Nova Transação",
          },
          Object {
            "icon": "fa-list",
            "id": "2",
            "onPress": [Function],
            "title": "Listar Transações",
          },
        ]
      }
    />
  </View>
</RCTSafeAreaView>
`;
